eduhelx_jupyterlab_prof/instructor_repo.py

```diff
@@ -1,49 +1,77 @@
 class InstructorClassRepo:
+    FIXED_REPO_ROOT = "eduhelx/{}-prof" # <class_name>
+    ORIGIN_REMOTE_NAME = "origin"
+    # Local branch
+    MAIN_BRANCH_NAME = "main"
+    # We have to do a merge to sync changes. We stage the merge on a separate branch
+    # to proactively guard against a merge conflict.
+    MERGE_STAGING_BRANCH_NAME = "__temp__/merge_{}-from-{}" # Formatted with the local head and tracking head commit hashes
+    ORIGIN_TRACKING_BRANCH = f"{ ORIGIN_REMOTE_NAME }/{ MAIN_BRANCH_NAME }"
+
     def __init__(self, course, assignments, current_path):
         self.course = course
         self.assignments = assignments
         self.current_path = os.path.realpath(current_path)
         
-        self.repo_root = self._compute_repo_root(self.course, self.current_path)
+        self.repo_root = self._compute_repo_root(self.course["name"], self.current_path)
         self.current_assignment = self._compute_current_assignment(self.assignments, self.repo_root, self.current_path)
     
+    @property
+    def current_assignment_path(self) -> Path | None:
+        if self.current_assignment is None: return None
+        return self.get_assignment_path(self.current_assignment)
+
     def get_assignment_path(self, assignment):
-        return os.path.join(self.repo_root, assignment["directory_path"])
+        return self.repo_root / assignment["directory_path"]
+    
+    def create_student_notebook(self):
+        if self.current_assignment is None:
+            raise NotInAnAssignmentException()
+        
+        assignment = self.current_assignment
+        master_notebook_path = self.current_assignment_path / assignment["master_notebook_path"]
+        student_notebook_path = self.current_assignment_path / assignment["student_notebook_path"]
+        dist_path = self.current_assignment_path / f"{ assignment['name'] }-dist"
+        student_notebook_dist_path = dist_path / "student" / master_notebook_path.name
 
-    @staticmethod
-    def _compute_repo_root(course, current_path):
-        # Make sure the path is a git repository and has a remote corresponding to the class repo remote
-        try: 
-            master_repo_remote = git.get_remote(name="origin", path=current_path)
-            repo_root = os.path.realpath(
-                git.get_repo_root(path=current_path)
-            )
-            if master_repo_remote != course["master_remote_url"]:
+        otter_assign(master_notebook_path, dist_path, no_pdfs=True)
+        shutil.move(student_notebook_dist_path, student_notebook_path)
+        shutil.rmtree(dist_path)
+    
+    @classmethod
+    def _compute_repo_root(cls, course_name, current_path: str | None=None):
+        """ Validates that user is in the repository root if current_path is provided """
+        repo_root = Path(cls.FIXED_REPO_ROOT.format(course_name.replace(" ", "_")))
+        if current_path is not None:
+            try:
+                Path(current_path).relative_to(os.path.realpath(repo_root))
```

Comment:
I think you should get the `realpath()` of `current_path` first as well before running `relative_to()` on it, so that you're sure you're not running `relative_to()` on a relative path. You might be able to guarantee that the input parameter is an absolute path in all cases _currently_, but perhaps in a future use of this method you try using it with a relative path and it fails.

Reply:
pathlib supports relative paths. using realpath here would just resolve symlinks which would be pointless.

Reply:
In [this](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to) documentation the ValueError message that's thrown when resolution is impossible is "ValueError: '<path1>' is not in the subpath of '<path2>' OR one path is relative and the other is absolute." That makes me think that both need to be absolute paths.

Reply:
ah good catch